fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java

```diff
@@ -170,163 +193,201 @@ public void testSearchQuantity_Quantity_NoDisplayUnit() throws Exception {
+        // And this is still a RANGE search.
+        assertSearchReturnsSavedResource("Range", "lt10.05||s");
         assertSearchReturnsSavedResource("Range", "lt11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_GT() throws Exception {
         assertSearchReturnsSavedResource("Range", "gt4||s");
         assertSearchReturnsSavedResource("Range", "gt5||s");
         assertSearchDoesntReturnSavedResource("Range", "gt10||s");
         assertSearchDoesntReturnSavedResource("Range", "gt11||s");
-        
-        assertSearchDoesntReturnSavedResource("Range", "le4||s");
-        assertSearchReturnsSavedResource("Range", "le5||s");
-        assertSearchReturnsSavedResource("Range", "le10||s");
-        assertSearchReturnsSavedResource("Range", "le11||s");
-        
-        assertSearchReturnsSavedResource("Range", "ge4||s");
-        assertSearchReturnsSavedResource("Range", "ge5||s");
-        assertSearchReturnsSavedResource("Range", "ge10||s");
-        assertSearchDoesntReturnSavedResource("Range", "ge11||s");
-        
-        assertSearchReturnsSavedResource("Range", "sa4||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa5||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa10||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_EB() throws Exception {
         assertSearchDoesntReturnSavedResource("Range", "eb4||s");
         assertSearchDoesntReturnSavedResource("Range", "eb5||s");
-        assertSearchDoesntReturnSavedResource("Range", "eb10||s");
+        // We use the range, so we actually return the value here. 
+        assertSearchReturnsSavedResource("Range", "eb10||s");
         assertSearchReturnsSavedResource("Range", "eb11||s");
     }
+
+    @Test
+    public void testSearchQuantity_Range_SA() throws Exception {
+        assertSearchReturnsSavedResource("Range", "sa4||s");
+        assertSearchDoesntReturnSavedResource("Range", "sa10||s");
+        assertSearchDoesntReturnSavedResource("Range", "sa10.0||s");
+        assertSearchDoesntReturnSavedResource("Range", "sa11||s");
+    }
     
+    @Test
+    public void testSearchQuantity_Range_GE() throws Exception {
+        assertSearchReturnsSavedResource("Range", "ge4||s");
+        assertSearchReturnsSavedResource("Range", "ge5||s");
+        // We're using the lowerbound to trigger the range search
+        assertSearchDoesntReturnSavedResource("Range", "ge10||s");
+        assertSearchDoesntReturnSavedResource("Range", "ge11||s");
+    }
+
+    @Test
+    public void testSearchQuantity_Range_LE() throws Exception {
+        assertSearchDoesntReturnSavedResource("Range", "le4||s");
+        assertSearchDoesntReturnSavedResource("Range", "le5.01||s");
+        // the upper bound is actually higher than the valueRange we inserted. 
+        assertSearchDoesntReturnSavedResource("Range", "le10||s");
```

Comment:
I disagree with this one too.  The target range is 5-10 which is le10....`le` should be looking for overlapping ranges and these range definitely overlap.

Reply:
great find. you are right. it was using lowerbound, it should have been using upperbound.